lib/ingram_micro/transmission.rb

```diff
@@ -8,19 +8,24 @@ class IngramMicro::Transmission
     'load-success' => 'inbound/BPXML-LoadSuccess.xsd',
     'return-receipt' => 'inbound/BPXML-ReturnReceipt.xsd',
     'ship-advice' => 'inbound/BPXML-ShipAdvice.xsd'
-  }
+  }.freeze
 
   attr_reader :errors, :transaction_name
 
-  def initialize(options={})
+  def initialize(options = {})
     @errors = []
   end
 
+  # schema_valid? compares an xml document (the result of self.xml_builder) to
+  # the appropriate xsd (XML Schema Definition).
+  # NOTE: the order of the fields matters when checking against the xsd files.
+  #   So, if in the order-header of a sales order, if customer_channel_type
```

Comment:
include `, but if customer_channel_type appears at end of xsd file, it will fail...`